akka-http2-support/src/test/scala/akka/http/impl/engine/http2/H2cUpgradeSpec.scala

```diff
@@ -47,31 +40,24 @@ Upgrade: h2c
 HTTP2-Settings: $settings
 
 """
-      val sink = Source.single(ByteString(upgradeRequest)).concat(Source.maybe)
-        .via(Tcp().outgoingConnection(binding.localAddress.getHostName, binding.localAddress.getPort))
-        .runWith(Sink.queue())
-      val response = sink.pull().futureValue.get.map(_.toChar).mkString
-      response should include("HTTP/1.1 101 Switching Protocols")
-      response should include("Upgrade: h2c")
-      response should include("Connection: upgrade")
-      val responseBody = sink.pull().futureValue.get
```

Comment:
In the failing case, probably this one pulled in the HTTP/1.1 headers plus the SETTINGS frame (coalesced by TCP), so the assertion on the SETTINGS frame type below failed.